daemon/daemon.go

```diff
@@ -1466,6 +1471,14 @@ func (daemon *Daemon) ImageService() ImageService {
 	return daemon.imageService
 }
 
+// ImageBackend returns an image-backend for Swarm and the distribution router.
+func (daemon *Daemon) ImageBackend() executorpkg.ImageBackend {
```

Comment:
It's rather unfortunate that we have this public method just so that we can create a router and a cluster on startup. The daemon here acts like an options object that carries around things used by other unrelated things

Reply:
Yes, I agree. I think we should make a pass at looking what's there, and if all of these are needed. I feel there's a lot of "ad-hoc" (oh, we need a new backend, just add something).